tests/spread/plugins/craft-parts/matter-sdk/snapcraft.yaml

```diff
@@ -0,0 +1,48 @@
+name: matter-lighting
+summary: Matter plugin test
+description: An lighting application to test the matter plugin.
+version: "1.0"
+
+base: core22
+
+grade: stable
+build-base: core22
+confinement: strict
+
+layout:
+  /mnt:
+    bind: $SNAP_COMMON/mnt
+
+apps:
+  matter-lighting:
+    daemon: simple
+    command: bin/lighting-app
+    install-mode: disable
+    plugs:
+      - network
+      - network-bind
+      - bluez
+      - avahi-control
+
+parts:
+  matter-sdk:
+    plugin: matter-sdk
+    matter-sdk-version: "1536ca20c5917578ca40ce509400e97b52751788" # use this commit with ptpython version fix; needs to be updated once matter sdk have a stable release
+
+  lighting:
+    plugin: nil
+    after: [matter-sdk]
+    override-build: |
+      # Source the Matter SDK environment variables
+      source ../../matter-sdk/build/matter-sdk-env.sh
+
+      # Build the lighting app for snapcraft spread testing purposes
+      cd ../../matter-sdk/build/examples/lighting-app/linux
+      gn gen out/build
+      ninja -C out/build
+      
+      ldd out/build/chip-lighting-app
+      
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp out/build/chip-lighting-app $CRAFT_PART_INSTALL/bin/lighting-app   
```

Comment:
should the matter plugin deal with this? Are all project different? I we cannot have nice reproduceable logic for a plugin we might need to sit down and discuss this a bit more.

Reply:
The reason here we enter the SDK directory `/matter-sdk/build` is to build an example application from it, as the SDK itself contains some example applications. 

For production, users will not build these example applications; they only need to run one command `source ../../matter-sdk/build/matter-sdk-env.sh` to set up the build environment, and then add their own logic.